src/SequenceManager.cc

```diff
@@ -243,6 +324,28 @@ SequenceName SequenceManager::make_sequence_name_from_label(gul14::string_view l
     return SequenceName{ name };
 }
 
+gul14::optional<SequenceManager::SequenceOnDisk> SequenceManager::parse_folder_name(
+    const std::filesystem::path& folder)
+{
+    const std::string str = unescape_filename_characters(folder.filename().string());
+
+    auto opening_bracket = str.rfind('[');
+    auto closing_bracket = str.find(']', opening_bracket);
+    if (opening_bracket != std::string::npos && closing_bracket == str.size() - 1)
```

Comment:
For the closing bracket you explicitly assume it at the end of the string, so precisely match `<name>[<uid>]`. I would prefer to change it to less than or equal

```C++
 if (... closing_bracket <= str.size() -1)
```

or

```C++
if (... closing_bracket < str.size())
```

But it is more a matter of taste. What so you think about it?

Reply:
That would mean that we accept folder names like `hello[0123456789abcdef]world`. But this would not work: Even if we load a sequence from such a folder, it would be saved in a folder called `hello[0123456789abcdef]` afterwards. We do not store the folder name anywhere, just its parts (the name and the UID).